web/src/components/profile/ProfileAddressLockedFields.tsx

```diff
@@ -2,12 +2,63 @@ import { ArrowTooltip } from "@/components/ArrowTooltip";
 import { Icon } from "@/components/njwds/Icon";
 import { AddressContext } from "@/contexts/addressContext";
 import { getMergedConfig } from "@/contexts/configContext";
-import { ReactElement, useContext } from "react";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { ReactElement, ReactNode, useContext } from "react";
 
 export const ProfileAddressLockedFields = (): ReactElement => {
   const { state } = useContext(AddressContext);
+  const { business } = useUserData();
 
   const Config = getMergedConfig();
+
+  const displayAddress = (): ReactNode => {
+    const businessLocation = business?.formationData.formationFormData.businessLocationType;
+    switch (businessLocation) {
+      case "NJ":
```

Comment:
[pebble] I think this `switch` could be further simplified. NJ and US addresses are rendered identically. Additionally, all 3 cases have the same logic for how `addressLine2` is conditionally displayed.

Doing this would also allow you to reduce the number of test cases you have to manage and account for. Right now, there's a describe for each case and you're covering these conditions:
- complete address is present
- missing addressLine2
- no address provided

That comes out to 9 test cases for what I'm seeing as 4 conditions.
- complete address
- missing addressLine2
- International address uses province
- no address provided